include/stellar/diagnostics/print.tpp

```diff
@@ -0,0 +1,207 @@
+    }
+
+    outStr << "  s^min        : " << statistics.smin << std::endl;
+    outStr << "  threshold    : " << statistics.threshold << std::endl;
+    outStr << "  distance cut : " << statistics.distanceCut << std::endl;
+    outStr << "  delta        : " << statistics.delta << std::endl;
+    outStr << "  overlap      : " << statistics.overlap << std::endl;
+    outStr << std::endl;
+}
+
+///////////////////////////////////////////////////////////////////////////////
+// Writes file name from options object to outStr
+template <typename TStream>
+void _writeFileNames(StellarOptions const & options, TStream & outStr)
+{
+//IOREV _notio_
+    outStr << "I/O options:" << std::endl;
+    outStr << "  database file   : " << options.databaseFile << std::endl;
+    outStr << "  query file      : " << options.queryFile << std::endl;
+    outStr << "  alphabet        : " << options.alphabet << std::endl;
+    outStr << "  output file     : " << options.outputFile << std::endl;
+    outStr << "  output format   : " << options.outputFormat << std::endl;
+    if (options.disableThresh != std::numeric_limits<size_t>::max())
+    {
+        outStr << "  disabled queries: " << options.disabledQueriesFile << std::endl;
+    }
+    outStr << std::endl;
+}
+
+///////////////////////////////////////////////////////////////////////////////
+// Calculates parameters from parameters in options object and from sequences and writes them to outStr
+template <typename TStringSet, typename TSize, typename TStream>
+void _writeMoreCalculatedParams(StellarOptions const & options,
+                                TSize const & refLen,
+                                TStringSet const & queries,
+                                TStream & outStr)
+{
+//IOREV _notio_
+    typedef typename Value<typename Value<TStringSet>::Type>::Type TAlphabet;
+
+    if (options.qgramAbundanceCut != 1)
+    {
+        outStr << "Calculated parameters:" << std::endl;
+    }
+
+    TSize queryLength = length(concat(queries));
+    if (options.qgramAbundanceCut != 1)
+    {
+        outStr << "  q-gram expected abundance : ";
+        outStr << queryLength / (double)((long)1 << (options.qGram << 1)) << std::endl;
+        outStr << "  q-gram abundance threshold: ";
+        outStr << _max(100, (int)(queryLength * options.qgramAbundanceCut)) << std::endl;
+        outStr << std::endl;
+    }
+
+    if (IsSameType<TAlphabet, Dna5>::VALUE || IsSameType<TAlphabet, Rna5>::VALUE)
+    {
+        // Computation of maximal E-value for this search
+
+        TSize maxLengthQueries = 0;
+
+        typename Iterator<TStringSet const>::Type queriesIt = begin(queries);
+        typename Iterator<TStringSet const>::Type queriesEnd = end(queries);
+        while (queriesIt != queriesEnd)
+        {
+            if (length(*queriesIt) > maxLengthQueries)
+            {
+                maxLengthQueries = length(*queriesIt);
+            }
+            ++queriesIt;
+        }
```

Comment:
This can be shortened to (requires `#include <algorithm>`):
```suggestion
        TSize maxLengthQueries = std::ranges::max(queries, {}, &seqan2::length);
```
I am not sure how `seqan2::length` is defined, if the above is not working the following should work:

```suggestion
        TSize maxLengthQueries = std::ranges::max(queries, {}, [](auto const& e) {
            return length(e);
        });
```

Reply:
Neither of these compiles (constraints not satisfied). Will leave as is